presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -110,33 +112,29 @@ public void finish()
         lookupJoiner.finish();
 
         if (!finishing && lookupSourceFactory.hasSpilled()) {
```

Comment:
Do we have a race here?

* one operator gets `finish()`
* second operator starts spilling

the ordering of these two actions has huge consequences (if they happen as I put them, the first operator will be counted among expected consumer counts, but will never show up).

Maybe such an ordering is not possible, but this should be asserted in the code. For example:

```
if (!finishing) {
 hasSpilled = lookupSourceFactory.tellTheFactoryIFinishedAndAskIfAnyoneHasSpilledOhAndTellItToPreventAnyFutureSpillsBtw(); 
 if (hasSpilled) {
    ....
```